### PR TITLE
Task/update build files

### PR DESCRIPTION
**Pull request by rhornung67**
Updates and house cleaning of build scripts and host-config files -- several new working versions of compilers and removal of older stuff.  There is still work to do to get more recent clang_coral compilers working with omp target, and also to get xl compilers running (both on Sierra/EA systems). Also, the plan is to rework the CMake stuff to avoid having to build two executables to support omp target. If possible, this will come in a later PR.

I want to get this merged in, so I can start another branch to updated RAJA and convert nested kernels to new 'kernel' constructs.

### CODE REVIEW

File: host-configs/blueos/clang_coral_2017_10_13-gpu.cmake [hunk 1]
```diff
@@ -15,7 +15,6 @@
 set(RAJA_COMPILER "RAJA_COMPILER_CLANG" CACHE STRING "")
 
 set(CMAKE_CXX_COMPILER "/usr/tce/packages/clang/clang-coral-2017.10.13/bin/clang++-gpu" CACHE PATH "")
-set(CMAKE_C_COMPILER "/usr/tce/packages/clang/clang-coral-2017.10.13/bin/clang-gpu" CACHE PATH "")
 
 set(CMAKE_CXX_FLAGS_RELEASE "-O3 -fno-vectorize" CACHE STRING "")
```
Comment (artv3) [inline host-configs/blueos/clang_coral_2017_10_13-gpu.cmake]: -fno-vectorize disables the vectorizer, do we want to have this in the release version?

Comment (davidbeckingsale) [reply host-configs/blueos/clang_coral_2017_10_13-gpu.cmake]: I think this may be to do with using OpenMP target?

Comment (rhornung67) [reply host-configs/blueos/clang_coral_2017_10_13-gpu.cmake]: @artv3 @davidbeckingsale with this version of the compiler, the results are incorrect without this flag. I'm working on trying newer versions of the compiler to see if the issue goes away. But, I'm having link issues....  Also, one of the RAJA CUDA view tests is failing in the develop branch. Trying to track that down first.

Comment (davidbeckingsale) [reply host-configs/blueos/clang_coral_2017_10_13-gpu.cmake]: That make sense. Could you add a comment to the host-config file about the error.

Comment (rhornung67) [reply host-configs/blueos/clang_coral_2017_10_13-gpu.cmake]: Sure. I'd like to remove that host-config and replace with more recent ones. But, unfortunately, that's the only one I can get thinks to link currently.

Comment (rhornung67) [reply host-configs/blueos/clang_coral_2017_10_13-gpu.cmake]: @davidbeckingsale and @artv3 I built and ran the perf suite without '-fno-vectorize' flag and results were correct and there was no significant performance difference. I removed it and pushed the change. I may need re-approval.

**Pull request merged**